File: 0509-fibonacci-number/0509-fibonacci-number.cpp
class Solution {
public:
    // int fib(int n) {
    //     if(n <= 1)
    //         return n;
    //     int F0 = 0;
    //     int F1 = 1;
    //     int F2 = 0;
    //     for(int i = 2; i <= n; i++){
    //         F2 = F1 + F0;
    //         F0 = F1;
    //         F1 = F2;
    //     }
    //     return F2;
    // }
    int fib(int n) {
       if(n <= 1) return n;
        
        return fib(n-1)+fib(n-2);
    }

};

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        F0 = 0
        F1 = 1

        if n == 0: return 0
        if n == 1: return 1

        for i in range(2, n+1):
            Fi = F0 + F1
            F0 = F1
            F1 = Fi

        return Fi
        